process/sync/baseSync.go

```diff
@@ -121,6 +123,15 @@ func (boot *baseBootstrap) loadBlocks(
 		}
 	}
 
+	firstNonceToDeleteFromStorer := highestNonceInStorer - lastBlocksToSkip + 1
+
+	for i := firstNonceToDeleteFromStorer; i <= highestNonceInStorer; i++ {
+		errNotCritical := boot.removeBlock(i, blockUnit, hdrNonceHashDataUnit)
+		if errNotCritical != nil {
+			log.Info(fmt.Sprintf("remove block with nonce %d: %s\n", i, err.Error()))
+		}
+	}
+
```

Comment:
I would split this entire method in 3 smaller methods:
1. computeHighestNonce
2. computeLatestCorrectBlock
3. removeAfterLatestCorrectBlock

Reply:
Refactored